crate/server/src/bootstrap_server/routes.rs

```diff
@@ -0,0 +1,329 @@
+use std::{path::PathBuf, sync::Arc};
+
+use actix_multipart::Multipart;
+use actix_web::{
+    post,
+    web::{Data, Json},
+    HttpRequest,
+};
+use cosmian_kms_utils::access::SuccessResponse;
+use futures::StreamExt;
+use openssl::pkcs12::Pkcs12;
+use serde::Deserialize;
+use tracing::warn;
+use url::Url;
+
+use crate::{
+    bootstrap_server::server::{BootstrapServer, BootstrapServerMessage},
+    config::DbParams,
+    database::redis::RedisWithFindex,
+    error::KmsError,
+    kms_error,
+    result::KResult,
+};
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct RedisFindexConfig {
+    pub url: String,
+    pub master_password: String,
+    pub findex_label: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct UrlConfig {
+    pub url: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct PathConfig {
+    pub path: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct PasswordConfig {
+    pub password: String,
+}
```

Comment:
There are only 2 crates common to both the server and the client

cosmian_kmip = { path = "../kmip" }
cosmian_kms_utils = { path = "../utils" }

So we have 3 choices: 
 - stuck these 4 structs into `util` (a synonym for "stuff we do not know what to do with and we put there and nobody will look at")
 - create a new crate, just for these 4 small structs 
 - have the client depend on the server -  does not look like a good idea to keep the client lean
 -accept the (almost) duplication of 16 lines
 
 I selected number 4. If someone can convince me one of the first 3 is better; no problem

Reply:
Okay, thanks for the detailed answer. I would have gone for the first option since `utils` already exists. But I understand your point, so let's go with your option :+1: